### ISSUE TITLE
Using Selected commit in Tree View

### ISSUE THREAD

**Issue by CodeMan99** [jonas/tig#377]
When in the main view you can hit `d` to go to the diff of that commit. While navigating up and down the commit id changes with the selection (the `[main] f49ef3bdccf8ea1caea3f15a626fe86f4642973c - commit 1 of 11952` part of the screen).

When in the tree view you have similar functionality, but if I hit `d` on a file object (blob?) it goes the selected commit of the main view. I would expect this to go the commit in the status: `[tree] 09d71b878ca6bc25dd5b3cef607409be2c6739d3 - file 8 of 19`.

Is this as simple as a tigrc change?

**Comment by jonas** [comment]
No, this would require changes to the file/blob view. But otherwise this makes sense.

As a workaround you can open then blame view to get access to the last diff that change the file by hitting `b` + `<Enter>`.

**Comment by CodeMan99** [comment]
Hm. Bit disappointing that it's not a tigrc change. I can see some long time users not liking the change.

Blame view is very good :sunglasses:

**Comment by jonas** [comment]
It is one of the areas where Tig is less consistent and one where
configuration is perhaps the answer. A first step is perhaps to enhance the
prompt so that you can configure what action you prefer. Something like:

```
# Open in split mode
bind blob d :vsplit diff %(blob-commit)
# Or maximized
bind blob d :open diff %(blob-commit)
```

I don't know what is most natural since we need to be both able to specify
the view to open, the way to open it ("auto" split, horizontal split,
vertical split) and pass additional state parameters, for example for blame
it would be commit + file + optional line number, for grep it is file +
line number.

**Comment by CodeMan99** [comment]
:+1: I personally got quite use to hitting "enter" (default action?) on an object bringing up a wide view. Now that you sometimes vertically split things I find myself thinking more before inputing commands. That pause is not very desirable when using a ncurses display.

I'd love to help with this, so if you find some ideas that you can delegate let me know. I have reasonable C/C++ skills.

**Comment by xemle** [comment]
+1
